### PR TITLE
Feature: Add online case library

### PR DESCRIPTION
**Pull request by IDKHTS**
Feature:
- home page online case resource list
- Case Resource model show case list and case detail
- Download Manager for Case Resource
   -  download and directly open model or project
   - Setting for Download Manager save path

### CODE REVIEW

File: src/app/ui/App.jsx [hunk 1]
```diff
@@ -154,6 +158,7 @@ class App extends PureComponent {
                         <Route path="/laser" component={Laser} />
                         <Route path="/cnc" component={Cnc} />
                         <Route path="/settings" component={Settings} />
+                        <Route path="/case-resouces" component={CaseResource} />
```
Comment (parachvte) [inline src/app/ui/App.jsx:161]: TODO: Rethink a better name

File: src/server/services/api/api-case-resrouces.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+const addPrefix = (prefix) => {
+    return function (request) {
+        if (request.url[0] === '/') {
+            request.url = prefix + request.url;
+        }
+
+        return request;
+    };
+};
+agent.use(addPrefix('http://45.79.80.155:8100'));
```
Comment (parachvte) [inline src/server/services/api/api-case-resrouces.js:17]: sub-domain

File: package.json [hunk 3]
```diff
@@ -116,6 +116,8 @@
     "cookie-parser": "1.4.6",
     "core-js": "3.6.5",
     "debug": "3.1.0",
+    "dexie": "^3.2.3",
```
Comment (parachvte) [inline package.json:119]: Use fixed version packages.

File: src/app/ui/App.jsx (hunk 1 above)
Comment (IDKHTS) [reply src/app/ui/App.jsx:161]: this route is not used, already remove this route

File: src/app/ui/pages/HomePage/Resources.jsx [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+import styles from './styles.styl';
+import QuickStart from './QuickStart';
+import api from '../../../api';
+import { MACHINE_SERIES, isDualExtruder } from '../../../constants/machines';
+import { actions as appGlobalActions } from '../../../flux/app-global';
+import { CaseConfigQuickStart } from './CaseConfig';
+import { renderModal } from '../../utils';
+import { DetailModalState } from '../../../constants/downloadManager';
+
+const Resources = (props) => {
```
Comment (parachvte) [inline src/app/ui/pages/HomePage/Resources.jsx:18]: Model Repository

Case Library

File: src/app/flux/app-global/index.js [hunk 5]
```diff
@@ -237,6 +260,14 @@ export const actions = {
             allPepaired: promptTasks.length === 0,
             results
         };
+    },
+
+    updateDownloadManangerSavedPath: (downloadManangerSavedPath) => dispatch => {
```
Comment (parachvte) [inline src/app/flux/app-global/index.js:265]: Avoid using app-global, this reducer will be removed later.

**Pull request merged**